### PR TITLE
[atip]Add new steps to check if many divs are the same

### PR DESCRIPTION
**Pull request by chenxix**
- Add new steps to save the specified div on page as pictures, and check if these divs are the same as others or not by picture similarity.

### CODE REVIEW

File: tools/atip/atip/web/web.py [hunk 1]
```diff
@@ -484,6 +488,105 @@ def fill_element_by_key(self, key, text, display=True):
...
+    def check_pic_same(self, pic1, pic2):
+        similarity = self.__cal_images_similar(pic1,pic2)
+        if similarity >= 99:
+           return True
+        else:
+           print "The similarity is: %s" % similarity
+           return False
+
+    def check_pic_different(self, pic1, pic2):
+        similarity = self.__cal_images_similar(pic1,pic2)
```
Comment (cicili) [inline tools/atip/atip/web/web.py]: why set __cal_images_similar as private?

Comment (chenxix) [inline tools/atip/atip/web/web.py]: some inconvenient ?

File: tools/atip/atip/common/common.py [hunk 2]
```diff
@@ -37,3 +40,52 @@ def __init__(self, app_config=None, app_name=None, timeout=2):
...
+           if pic_similarity >= string.atoi(similarity):
+              return True
+           else:
+              print "The similarity is: %s" % pic_similarity
+              return False
+        except Exception as e:
+           print "Check similarity failed: %s" % e
+           return False
+
+    def check_pic_different(self, pic1, pic2, similarity):
```
Comment (cicili) [inline tools/atip/atip/common/common.py]: update func name to "get_pic_similarity()"

File: tools/atip/atip/common/common.py [hunk 3]
```diff
@@ -37,3 +40,52 @@ def __init__(self, app_config=None, app_name=None, timeout=2):
...
+        pic_similarity = self.cal_images_similar(pic1,pic2)
+        if pic_similarity >= string.atoi(similarity):
+           print "The similarity is: %s" % pic_similarity
+           return False
```
Comment (cicili) [inline tools/atip/atip/common/common.py]: here return the pics similarity is ok

**Comment by cicili** [comment]
LGTM.

File: tools/atip/atip/common/common.py [hunk 4]
```diff
@@ -37,3 +40,52 @@ def __init__(self, app_config=None, app_name=None, timeout=2):
...
+    #standardize the image 
+    def make_regalur_image(self,img, size = (256, 256)):
+        return img.resize(size).convert('RGB')
+
+    def hist_similar(self,lh, rh):
+        assert len(lh) == len(rh)
+        return sum(1 - (0 if l == r else float(abs(l - r))/max(l, r)) for l, r in zip(lh, rh))/len(lh)
+
+    #Images similarity calculation
+    def cal_images_similar(self,img1,img2):
```
Comment (Honry) [inline tools/atip/atip/common/common.py]: There should be a white space after the comma, please unify the format.

File: tools/atip/atip/web/web.py [hunk 5]
```diff
@@ -477,6 +481,83 @@ def click_element_coords(self, x, y, key, display=True):
...
+    #Save the specified element as a single picture
+    def save_div_as_picture(self, key, element_pic, width=0, height=0):
+        try:
+           page_pic = "page.png"
+           self.driver.get_screenshot_as_file(page_pic)
+           self.picture_list.append(page_pic)
+           ratio = self.calculate_resolution_ratio(page_pic)
+           self.convert_pic(page_pic, ratio)
+           box = self.calculate_element_location(key, width, height)
+           self.crop_pic(page_pic, element_pic, box)
```
Comment (Honry) [inline tools/atip/atip/web/web.py]: Still self? You have moved "crop_pic" to common folder. Please also update others. We should have unit test for each new added function.

Comment (chenxix) [inline tools/atip/atip/web/web.py]: The crop_pic() function is inherited from parent class common.APP.

Comment (chenxix) [inline tools/atip/atip/web/web.py]: I have checked the added function on android platform, and the result is OK.

**Comment by Honry** [comment]
LGTM

**Pull request merged**